vector/src/test/java/im/vector/app/features/location/LocationDataTest.kt

```diff
@@ -28,19 +28,22 @@ import org.matrix.android.sdk.api.session.room.model.message.MessageLocationCont
 class LocationDataTest {
     @Test
     fun validCases() {
-        parseGeo("geo:12.34,56.78;13.56") shouldBeEqualTo
+        parseGeo("geo:12.34,56.78;u=13.56") shouldBeEqualTo
                 LocationData(latitude = 12.34, longitude = 56.78, uncertainty = 13.56)
         parseGeo("geo:12.34,56.78") shouldBeEqualTo
                 LocationData(latitude = 12.34, longitude = 56.78, uncertainty = null)
         // Error is ignored in case of invalid uncertainty
-        parseGeo("geo:12.34,56.78;13.5z6") shouldBeEqualTo
+        parseGeo("geo:12.34,56.78;u=13.5z6") shouldBeEqualTo
                 LocationData(latitude = 12.34, longitude = 56.78, uncertainty = null)
-        parseGeo("geo:12.34,56.78;13. 56") shouldBeEqualTo
+        parseGeo("geo:12.34,56.78;u=13. 56") shouldBeEqualTo
                 LocationData(latitude = 12.34, longitude = 56.78, uncertainty = null)
         // Space are ignored (trim)
-        parseGeo("geo: 12.34,56.78;13.56") shouldBeEqualTo
+        parseGeo("geo: 12.34,56.78;u=13.56") shouldBeEqualTo
+                LocationData(latitude = 12.34, longitude = 56.78, uncertainty = 13.56)
+        parseGeo("geo:12.34,56.78; u=13.56") shouldBeEqualTo
                 LocationData(latitude = 12.34, longitude = 56.78, uncertainty = 13.56)
-        parseGeo("geo:12.34,56.78; 13.56") shouldBeEqualTo
+        // missing "u=" for uncertainty is ignored
+        parseGeo("geo:12.34,56.78;13.56") shouldBeEqualTo
```

Comment:
I would have create a dedicated test for this case, since this is not a fully valid case. (but we are already testing invalid uncertainty here :/)

Reply:
Thanks!